avalon/maya/pipeline.py

```diff
@@ -447,6 +447,14 @@ def ls():
 
     for container in sorted(containers):
         data = parse_container(container)
+
+        # Hierarchical container support
+        parent = cmds.listSets(object=container) or []
+        for node in parent:
+            if node in containers:
+                data["parent"] = node
+                break
```

Comment:
I end up moving this logic from `config.collect_container_metadata` to here, since I thought the *hierarchy view* feature would apply to everyone, so the attribute `parent` which is required should also inside the `core`.

Reply:
Sure. The only thing I am a bit wary about is whether being present in the container is the functionality that will be the same basis for parenting for "everyone" ever. However, I do feel this is a good basis. Plus a bit about the speed of things too - since it involves a bit more queries.

Could we maybe optimize this query a bit? Should `containers` maybe be a `set` so the lookup becomes `O(1)` instead of `O(n)`.

How does this differ in speed for big scenes?

Reply:
Since I have changed to use different implementation, there should not have any over head on normal `ls()` query. :)

Reply:
I have changed to use `set` in some place in this implementation :)